runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -1916,30 +1916,29 @@ J9::TransformUtil::createDiamondForCall(TR::Optimization* opt, TR::TreeTop *call
 void J9::TransformUtil::removePotentialOSRPointHelperCalls(TR::Compilation* comp, TR::TreeTop* start, TR::TreeTop* end)
    {
    TR_ASSERT(start->getEnclosingBlock() == end->getEnclosingBlock(), "Does not support range across blocks");
+   TR_ASSERT(comp->supportsInduceOSR() && comp->isOSRTransitionTarget(TR::postExecutionOSR) && comp->getOSRMode() == TR::voluntaryOSR,
```

Comment:
do we also need to check that the OSR infrastructure hasn't been removed? it doesn't make sense to do this after that point I don't think since we have made a lot of complex decisions.

Reply:
`comp->supportsInduceOSR()` does the check.